src/main/java/com/socrata/api/Soda2Producer.java

```diff
@@ -76,6 +78,7 @@ public static final Soda2Producer newProducerWithRequestId(final String url, Str
     public Soda2Producer(HttpLowLevel httpLowLevel)
     {
         super(httpLowLevel);
+        factory = new JsonFactory();
```

Comment:
This class needs the MappingJsonFactory.  You will need to either get that from ObjectMapper or explicitly create one.  You can see test failures in Soda2ProducerTest if you just use the plain JsonFactory.